pages/api/events/upcoming.ts

```diff
@@ -0,0 +1,44 @@
+import dbConnect from '@/lib/dbConnect';
+import VolunteerEvents from 'bookem-shared/src/models/VolunteerEvents';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  // Get request method
+  const { method } = req;
+
+  switch (method) {
+    /**
+     * @route GET /api/events/upcoming
+     * @desc Get all events in the future
+     * @res QueriedVolunteerEventData[]
+     */
+    case 'GET':
+      try {
+        await dbConnect();
+
+        // Select all events where eventDate > today order by progamDate ascending
+        const events = await VolunteerEvents.find({
+          startDate: { $gt: new Date() },
+        })
+          .sort({ startDate: 1 })
+          .limit(5);
```

Comment:
This is the corrected version. Before it was `eventDate` which does not exist in VolunteerEvents model.